src/main/java/org/zone/commands/structure/region/flags/messages/greetings/ZoneFlagGreetingsViewCommand.java

```diff
@@ -46,8 +47,11 @@
     @Override
     public @NotNull CommandResult run(CommandContext commandContext, String... args) {
         Zone zone = commandContext.getArgument(this, ZONE_VALUE);
-        GreetingsFlag greetingsFlag = zone.getFlag(FlagTypes.GREETINGS).orElse(new GreetingsFlag());
-        Component message = greetingsFlag.getMessage().orElse(Messages.getNoMessageSet());
+        Optional<GreetingsFlag> opGreetingsFlag = zone.getFlag(FlagTypes.GREETINGS);
+        if (opGreetingsFlag.isEmpty()) {
+            return CommandResult.error(Messages.getNoMessageSet());
+        }
+        Component message = opGreetingsFlag.get().getMessage().orElse(Messages.getNoMessageSet());
```

Comment:
Perhaps the greeting message flag is wrong as the message should always be present, therefore ``getMessage`` should not return ``Optional``

Reply:
fixed